src/audio/types.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include <QtDebug>
+
+#include "util/assert.h"
+#include "util/optional.h"
+#include "util/types.h"
+
+// Various properties of digital PCM audio signals and streams.
+//
+// An audio signal or stream contains samples for multiple
+// channels sampled at discrete times.
+//
+// The channel layout (optional) assigns meaning to the
+// different channels of a signal.
+//
+// The sample layout defines how subsequent samples from
+// different channels are represented and stored in memory.
+
+namespace mixxx {
+
+namespace audio {
+
+enum class ChannelLayout {
+    Mono,     // 1 channel
+    DualMono, // 2 channels with identical signals
+    Stereo,   // 2 independent channels left/right
+    // ...to be continued...
+};
+
+typedef std::optional<ChannelLayout> OptionalChannelLayout;
+
+QDebug operator<<(QDebug dbg, ChannelLayout arg);
+
+class ChannelCount {
+  private:
+    static constexpr SINT kValueDefault = 0;
+
+  public:
+    static constexpr SINT kValueMin = 1;   // lower bound (inclusive)
+    static constexpr SINT kValueMax = 255; // upper bound (inclusive, 8-bit unsigned integer)
+
+    static constexpr ChannelCount min() {
+        return ChannelCount(kValueMin);
+    }
+    static constexpr ChannelCount max() {
+        return ChannelCount(kValueMax);
+    }
+
+    static ChannelCount fromLayout(ChannelLayout layout) {
+        switch (layout) {
+        case ChannelLayout::Mono:
+            return ChannelCount(1);
+        case ChannelLayout::DualMono:
+            return ChannelCount(1);
```

Comment:
I think this should return 2. If other code used this to determine where to read or write in a dual mono buffer, I think the results would be incorrect.